From 1e52f7155bcaf6a73a32c5d8b17d450d060abf82 Mon Sep 17 00:00:00 2001
Message-Id: <1e52f7155bcaf6a73a32c5d8b17d450d060abf82.1687727967.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1687727967.git.sharlatanus@gmail.com>
References: <cover.1687727967.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Wed, 21 Jun 2023 21:57:44 +0100
Subject: [PATCH 4/9] gnu: missfits: Fix build.

* gnu/packages/astronomy.scm (missfits): Fix build.
[source]: Swap to 'git-fetch.
[arguments]: Add CPPFLAGS to pass build with GCC 10.
---
 gnu/packages/astronomy.scm | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 326a9a170e..4c869fc295 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -2653,12 +2653,21 @@ (define-public missfits
     (version "2.8.0")
     (source
      (origin
-       (method url-fetch)
-       (uri (string-append "https://www.astromatic.net/download/missfits/"
-                           "missfits-" version ".tar.gz"))
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/astromatic/missfits")
+             (commit version)))
        (sha256
-        (base32 "04jrd7fsvzr14vdmwgj2f6v97gdcfyjyz6jppml3ghr9xh12jxv5"))))
+        (base32 "12ndvrr3l5j7ph2i5f3qf0wqmv5ymsyjzxnnypqajsvliw72iprh"))
+       (file-name (git-file-name name version))))
     (build-system gnu-build-system)
+    (arguments
+     (list
+      #:configure-flags
+      #~(list
+         ;; Address this link error:
+         ;; ld: ... multiple definition of ... first defined here
+         "CPPFLAGS=-O2 -g -fcommon")))
     (home-page "https://www.astromatic.net/software/missfits")
     (synopsis "FITS files Maintenance program")
     (description
-- 
2.40.1

